xarray/core/variable.py

```diff
@@ -492,6 +495,46 @@ def _broadcast_indexes_advanced(self, key):
         key = VectorizedIndexer(variable.data for variable in variables)
         return dims, key
 
+    def _broadcast_indexes_dask_pointwise(self, key):
+        if any(not isinstance(k, (Variable, slice)) for k in key):
```

Comment:
Yes, indeed, that should totally work. I don't have time to do that now but probably later this week.

Actually, this gives us a path to supporting vectorized indexing via the same code path for dask and numpy. The only differences:
- For dask.array, we need to explicitly broadcast and flatten indexers. (This could also be fixed upstream.)
- For numpy, we don't have `vindex`, so for now we need to convert to `slice()` objects to arrays to keep things predictable. But I think there is a pure NumPy `vindex` implementation sitting around somewhere, I could look it up. (And/or encourage @seberg to finish up https://github.com/numpy/numpy/pull/6256.)

We will still want to keep the separate OuterIndexer path. That allows dask to make some nice optimizations.